src/map/unit.cpp

```diff
@@ -2784,10 +2812,12 @@ static int unit_attack_timer_sub(struct block_list* src, int tid, t_tick tick)
 			unit_stop_walking(src,1);
 
 		if(md) {
-			//First attack is always a normal attack
-			if(md->state.skillstate == MSS_ANGRY || md->state.skillstate == MSS_BERSERK) {
-				if (mobskill_use(md,tick,-1))
+			// Berserk skills can replace normal attacks except for the first attack (state not berserk yet)
```

Comment:
According to the check below the state is already berserk. So why does your comment say "not berserk yet"?

Reply:
I thought the original comment doesn't really explain how this code makes the first attack being a normal attack, so I added the explanation that this is because it's not in berserk state yet so it would skip this code part.
Seems it's still not clear enough.

Reply:
Comment improved.